TAO/TAO_IDL/include/ast_expression.h

```diff
@@ -279,10 +287,11 @@ class TAO_IDL_FE_Export AST_Expression
   // Evaluate then store value inside this AST_Expression.
   void evaluate (EvalKind ek);
 
-  // Compare two AST_Expressions.
-
+  /// Compare two AST_Expressions.
+  ///{
   bool operator== (AST_Expression *vc);
-  long compare (AST_Expression *vc);
+  bool compare (AST_Expression *vc);
```

Comment:
Was this previously a `strcmp` style return value (negative, 0, positive)?
If it's going to return `bool` would a different name be better?

Reply:
> Was this previously a strcmp style return value (negative, 0, positive)?

[No](https://github.com/DOCGroup/ACE_TAO/blob/943dd53bd48bf4a16c02f98d217c59bbb4b78ea7/TAO/TAO_IDL/ast/ast_expression.cpp#L2891)

> If it's going to return bool would a different name be better?

I think I added a `operator==` for it.

Reply:
Maybe, but I left it there for some level of compatibility.